tests/conftest.py

```diff
@@ -0,0 +1,255 @@
+import pytest
+import json
+import logging
+import os
+import time
+import yaml
+
+from oc.openshift_ops import OCP
+from ocs.exceptions import CommandFailed, CephHealthException
+from ocs.utils import create_oc_resource, apply_oc_resource
+from ocsci import config
+from ocsci.config import RUN, ENV_DATA, DEPLOYMENT
+from utility import templating
+from utility.aws import AWS
+from utility.retry import retry
+from utility.utils import destroy_cluster, run_cmd, get_openshift_installer, get_openshift_client, is_cluster_running
+from ocs.parallel import parallel
+
+log = logging.getLogger(__name__)
+
+
+def cluster_teardown():
+    log.info("Destroying the test cluster")
+    destroy_cluster(ENV_DATA['cluster_path'])
+    log.info("Destroying the test cluster complete")
+
+
+@pytest.fixture(scope="session", autouse=True)
+def cluster(request):
+    log.info("Running OCS basic installation")
+    cluster_path = ENV_DATA['cluster_path']
+    # Add a finalizer to teardown the cluster after test execution is finished
+    if config.teardown:
+        request.addfinalizer(cluster_teardown)
+        log.info("Will teardown cluster because --teardown was provided")
+    # Test cluster access and if exist just skip the deployment.
+    if is_cluster_running(cluster_path):
```

Comment:
Let's execute it only when user requests it to by specifying in the command line (i.e. --deploy_cluster)